Gems/Terrain/Assets/Shaders/Terrain/TerrainDetailHelpers.azsli

```diff
@@ -257,46 +257,42 @@ void BlendDetailMaterial(inout DetailSurface surface, uint material1, uint mater
     }
 }
 
 /*
 Populates a DetailSurface with material data gathered form the 4 nearest samples to detailMaterialIdUv. The weight
 of each detail material's contribution is calculated based on the distance to the center point for that sample (for
-instance, if detailMaterialIdUv falls perfectly inbetween all 4 samples, then each sample will be weighed at 25%).
+instance, if detailMaterialIdUv falls perfectly in-between all 4 samples, then each sample will be weighed at 25%).
 Each sample can have two different detail materials defined with a blend value to determine their relative contribution.
-The detailUv is used for samping the textures of each detail material.
+The detailUv is used for sampling the textures of each detail material.
 */
 bool GetDetailSurface(inout DetailSurface surface, float2 detailMaterialIdUv, float2 detailUv)
 {
     float2 textureSize;
     TerrainSrg::m_detailMaterialIdImage.GetDimensions(textureSize.x, textureSize.y);
 
-    float2 detailMaterialIdCoord = detailMaterialIdUv * textureSize + 0.5;
-    float2 detailMaterailIdUv = detailMaterialIdCoord;
-    float2 gatherWeight = frac(detailMaterialIdCoord);
+    float2 detailMaterialIdCoord = detailMaterialIdUv * textureSize; // uv -> pixel coordinate
 
-    // Adjust the gather weight for better interpolation. This helps avoid diamond-shaped artifacts in binlinear filtering.
-    gatherWeight = 3.0 * gatherWeight * gatherWeight - 2.0 * gatherWeight * gatherWeight * gatherWeight;
+    // detailMaterialIdCoord could be negative, so add textureSize to ensure it is positive
+    detailMaterialIdCoord += textureSize; 
+
+    // The detail material id texture wraps since the "center" point can be anywhere in the texture, so mod by texturesize
+    int2 detailMaterailIdTopLeft = int2(detailMaterialIdCoord) % textureSize;
+    int2 detailMaterailIdBottomRight = (int2(detailMaterialIdCoord) + 1) % textureSize;
 
     // Using Load() to gather the nearest 4 samples (Gather4() isn't used because of precision issues with uvs).
-    uint4 s1 = TerrainSrg::m_detailMaterialIdImage.Load(int3((detailMaterialIdCoord + int2(0, 1) + textureSize) % textureSize, 0));
-    uint4 s2 = TerrainSrg::m_detailMaterialIdImage.Load(int3((detailMaterialIdCoord + int2(1, 1) + textureSize) % textureSize, 0));
-    uint4 s3 = TerrainSrg::m_detailMaterialIdImage.Load(int3((detailMaterialIdCoord + int2(1, 0) + textureSize) % textureSize, 0));
-    uint4 s4 = TerrainSrg::m_detailMaterialIdImage.Load(int3((detailMaterialIdCoord + int2(0, 0) + textureSize) % textureSize, 0));
+    uint4 s1 = TerrainSrg::m_detailMaterialIdImage.Load(int3(detailMaterailIdTopLeft.x, detailMaterailIdBottomRight.y, 0));
+    uint4 s2 = TerrainSrg::m_detailMaterialIdImage.Load(int3(detailMaterailIdBottomRight, 0));
+    uint4 s3 = TerrainSrg::m_detailMaterialIdImage.Load(int3(detailMaterailIdBottomRight.x, detailMaterailIdTopLeft.y, 0));
+    uint4 s4 = TerrainSrg::m_detailMaterialIdImage.Load(int3(detailMaterailIdTopLeft, 0));
```

Comment:
Suggestion:  To go along with the previous comment, an alternate way I'd find this a little bit more readable is if it worked its way from top left to bottom right, or at least spelled it out in all cases, like this:
```
int3(TopLeft.x,     TopLeft.y, 0)
int3(TopLeft.x,     BottomRight.y, 0)
int3(BottomRight.x, TopLeft.y, 0)
int3(BottomRight.x, BottomRight.y, 0)
```